client/filejsonstore/filejsonstore.go

```diff
@@ -0,0 +1,121 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/theupdateframework/go-tuf/util"
+)
+
+// ErrNotJSON is returned when a metadata operation is attempted to be
+// performed against a file that does not seem to be a JSON file
+// (e.g. does not end in .json, case sensitive).
+var ErrNotJSON = errors.New("file is not in JSON format")
+
+// FileJSONStore represents a local metadata cache relying on raw JSON files
+// as retrieved from the remote repository.
+type FileJSONStore struct {
+	baseDir string
+}
+
+// NewFileJSONStore returns a new metadata cache, implemented using raw JSON
+// files, stored in a directory provided by the client.
+// If the provided directory does not exist on disk, it will be created.
+// The provided metadata cache is not safe for concurrent access, if
+// concurrent access safety is requires, wrap local store in a
+// ConcurrentLocalStore.
+func NewFileJSONStore(baseDir string) (*FileJSONStore, error) {
+	return newImpl(baseDir, true)
+}
+
+func newImpl(baseDir string, recurse bool) (*FileJSONStore, error) {
+	var f = FileJSONStore{
+		baseDir: baseDir,
+	}
+	var err error
+
+	// Does the directory exist?
+	fi, err := os.Stat(baseDir)
```

Comment:
Perhaps we should check permissions on the directory and the files inside (in `GetMeta`) if it already exists, similar to how SSH validates keys on disk.

Reply:
Sure, I'll add checks to make sure only `user` (owner) have write permission to the directory and any files in it.